lib/runner.js

```diff
@@ -266,13 +266,13 @@ class Runner extends Emittery {
 		return result;
 	}
 
-	async runHooks(tasks, contextRef, titleSuffix) {
+	async runHooks(tasks, contextRef, testArgs = [], titleSuffix) {
 		const hooks = tasks.map(task => new Runnable({
 			contextRef,
 			failWithoutAssertions: false,
 			fn: task.args.length === 0 ?
-				task.implementation :
-				t => task.implementation.apply(null, [t].concat(task.args)),
+				t => task.implementation.apply(null, [t].concat(testArgs)) :
+				t => task.implementation.apply(null, [t].concat(task.args).concat(testArgs)),
```

Comment:
We should clean this up and use rest syntax: `task.implementation.call(null, t, ...testArgs)`.